policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py

```diff
@@ -0,0 +1,34 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment(Variable):
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut social security benefit adjustment"
+    definition_period = YEAR
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.irs.social_security.taxability
+        income_base_amount = p.threshold.lower
+        ss_agi_frac = p.rate.lower
+        filing_status = tax_unit("filing_status", period)
+        base_amount = income_base_amount[filing_status]
+
+        c = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        ss_rate = c.rate
+        filing_status = tax_unit("filing_status", period)
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
+        income_threshold = c.income_threshold[filing_status]
+
```

Comment:
It seems like the comparison is happening between the `taxable_social_security` and (25% of `social_security` or 25% of a IRS 86 (b) (1) - this is described in tax_unit_taxable_social_security and we need to replicate it (in the irs folder)```

    def formula(tax_unit, period, parameters):
        ss = parameters(period).gov.irs.social_security.taxability
        gross_ss = tax_unit("tax_unit_social_security", period)

        ss_fraction = ss.rate.lower * gross_ss
        modified_agi_plus_half_ss = (
            tax_unit("taxable_ss_magi", period) + ss_fraction
        )
        filing_status = tax_unit("filing_status", period)

        base_amount = ss.threshold.lower[filing_status]

        excess_over_base = max_(0, modified_agi_plus_half_ss - base_amount)

```
![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/11012d88-0855-4e2a-a354-e6ce003f9ff1)